### PR TITLE
feat: added headers to the onResponse callback.

### PR DESCRIPTION
**Pull request by EladIsraeli**

### CODE REVIEW

**Comment by EladIsraeli** [comment]
Resolves https://github.com/mcollina/autocannon/issues/361

**Review by mcollina — COMMENTED** [review]
Could you add docs?

**Comment by EladIsraeli** [comment]
Yeah sure, I added docs let me know if there is anything related to the docs needed here

File: lib/requestIterator.js [hunk 1]
```diff
@@ -106,9 +106,9 @@ RequestIterator.prototype.rebuildRequest = function () {
   return data
 }
 
-RequestIterator.prototype.recordBody = function (request, status, body) {
+RequestIterator.prototype.recordBody = function (request, status, body, headers) {
   if (request && typeof request.onResponse === 'function') {
-    request.onResponse(status, body, this.context)
+    request.onResponse(status, body, this.context, headers)
```
Comment (mcollina) [inline lib/requestIterator.js:111]: Can you move the array-to-object logic within this if? We can do it only if `onResponse` is set.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: lib/requestIterator.js (hunk 1 above)
Comment (EladIsraeli) [reply lib/requestIterator.js:111]: Sounds good! it could spare us an unnecessary conversation.

File: lib/requestIterator.js [hunk 2]
```diff
@@ -106,9 +106,19 @@ RequestIterator.prototype.rebuildRequest = function () {
...
 }
 
-RequestIterator.prototype.recordBody = function (request, status, body) {
+RequestIterator.prototype.recordBody = function (request, status, body, headers) {
   if (request && typeof request.onResponse === 'function') {
-    request.onResponse(status, body, this.context)
+    request.onResponse(status, body, this.context, toHeaderKeyValue(headers || []))
+  }
+
+  function toHeaderKeyValue (headersArray) {
```
Comment (mcollina) [inline lib/requestIterator.js:114]: Can you move this to a top level function?

File: lib/requestIterator.js [hunk 3]
```diff
@@ -106,9 +106,19 @@ RequestIterator.prototype.rebuildRequest = function () {
...
-    request.onResponse(status, body, this.context)
+    const toHeaderKeyValue = (headersArray) => {
+      const headersKeyValue = {}
+
+      for (let i = 0; i < headersArray.length - 1; ++i) {
+        headersKeyValue[headersArray[i]] = headersArray[i + 1]
+      }
+
+      return headersKeyValue
+    }
```
Comment (mcollina) [inline lib/requestIterator.js:119]: by top level I mean to not allocate a closure and call a function defined outside of this one.

**Pull request merged**